### PR TITLE
Add tests for DesployRadiatorController

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?

Adding tests for `DesployRadiatorController`.
I also slightly refactored `DesployRadiatorController` to be able to record requests made to teamcity and riff-raff
## What is the value of this?

I am thinking about adding an endpoint to notify build/deploy changes
I will probably need to make some changes in order to reuse some code. I prefer to have tests in place before to do so.
## Request for comment

@OliverJAsh

### CODE REVIEW

File: admin/test/controllers/DeploysRadiatorControllerTest.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      status(response) should be(OK)
+
+      val jsonResponse = contentAsJson(response)
+      ((jsonResponse \ "response").as[JsArray]).value.size should be(pageSize)
+      (jsonResponse \ "status").as[String] should be("ok")
+      (jsonResponse \ "response" \\ "projectName").map(_.as[String]).distinct should equal(List(projectName))
+      (jsonResponse \ "response" \\ "projectName").map(_.as[String]).distinct should equal(List(projectName))
+      (jsonResponse \ "response" \\ "stage").map(_.as[String]).distinct should equal(List(stage))
+    }
+    "returns 200 with no results when project and stage exists" in {
```
Comment (OliverJAsh) [inline admin/test/controllers/DeploysRadiatorControllerTest.scala]: Should this be “when project and stage do not exist”?

File: admin/test/controllers/DeploysRadiatorControllerTest.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  def getUri: Uri = throw new NotImplementedError()
+  def getHeader(name: String): String = throw new NotImplementedError()
+  def getHeaders(name: String): util.List[String] = throw new NotImplementedError()
+  def getHeaders: FluentCaseInsensitiveStringsMap = throw new NotImplementedError()
+  def isRedirected: Boolean = throw new NotImplementedError()
+  def getCookies = throw new NotImplementedError()
+  def hasResponseStatus: Boolean = throw new NotImplementedError()
+  def hasResponseHeaders: Boolean = throw new NotImplementedError()
+  def hasResponseBody: Boolean = throw new NotImplementedError()
+}
```
Comment (OliverJAsh) [inline admin/test/controllers/DeploysRadiatorControllerTest.scala]: Just so I understand properly, is this used for mocking the response object?

**Comment by OliverJAsh** [comment]
Nice one! :+1:

Btw the build is failing.

File: admin/test/controllers/DeploysRadiatorControllerTest.scala (hunk 1 above)
Comment (TBonnin) [inline admin/test/controllers/DeploysRadiatorControllerTest.scala]: Yep. I will fix

File: admin/test/controllers/DeploysRadiatorControllerTest.scala (hunk 2 above)
Comment (TBonnin) [inline admin/test/controllers/DeploysRadiatorControllerTest.scala]: Exactly

**Comment by TBonnin** [comment]
Teamcity is failling because it doesn't have access to the riff-raff and teamcity properties

**Pull request merged**